rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependency.java

```diff
@@ -116,6 +116,8 @@ public class AddDependency extends ScanningRecipe<AddDependency.Scanned> {
     @Nullable
     Boolean acceptTransitive;
 
+    transient boolean requiresAnotherCycle = true;
```

Comment:
In a first iteration I was planning on having this transient field (initially `false`) which the recipe in the edit phase sets to `true` if it detects any source files which match `onlyIfUsing` so that a second cycle gets run.

A more proper fix would be a dedicated API. One idea would be to add an `ExecutionContext` parameter to the existing `Recipe#causesAnotherCycle()` method, which would allow the recipe to store something in the `ExecutionContext`. Another idea would be to add a new API to `RecipeRunCycle` (can be obtained via `ExecutionContext#getCycleDetails()`) or directory on `ExecutionContext`. I am thinking something like a `requestAnotherCycle()`. But the existing  `Recipe#causesAnotherCycle()` would likely have to remain for declarative recipes.

Depending on the approach, some additional changes may be required to the recipe scheduler, as it currently expects the recipe requesting another cycle to also have performed changes in the current cycle. This may not always be the case.